smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/HttpProtocolTestGenerator.java

```diff
@@ -217,6 +246,99 @@ private void generateRequestTest(OperationShape operation, HttpRequestTestCase t
         });
     }
 
+    private void generateServerRequestTest(OperationShape operation, HttpRequestTestCase testCase) {
+        Symbol operationSymbol = symbolProvider.toSymbol(operation);
+
+        // Lowercase all the headers we're expecting as this is what we'll get.
+        Map<String, String> headers = testCase.getHeaders().entrySet().stream()
+                .map(entry -> new Pair<>(entry.getKey().toLowerCase(Locale.US), entry.getValue()))
+                .collect(MapUtils.toUnmodifiableMap(Pair::getLeft, Pair::getRight));
+        String queryParameters = Node.prettyPrintJson(buildQueryBag(testCase));
+        String headerParameters = Node.prettyPrintJson(ObjectNode.fromStringMap(headers));
+        String body = testCase.getBody().orElse(null);
+
+        String testName = testCase.getId() + ":ServerRequest";
+        testCase.getDocumentation().ifPresent(writer::writeDocs);
+        writer.openBlock("it($S, async () => {", "});\n", testName, () -> {
+            // TODO: use the symbol provider when it's ready
+            String serviceName = StringUtils.capitalize(service.getId().getName());
+            String operationName = StringUtils.capitalize(operation.getId().getName());
+            Symbol inputType = operationSymbol.expectProperty("inputType", Symbol.class);
+
+            // Declare "r" to capture the deserialized input
+            writer.write("let r: any;");
+
+            // This doesn't actually satisfy the service's server interface since it only applies to one operation
```

Comment:
Do we not have access to a mocking facility that will satisfy the type checker and remove the need to do this manual state capture for verification?

Reply:
I updated to use a mock function there. I can't really get rid of casting, but I did rework it so that we're making better use of the type checker. Now the test service is defined as a `Partial` variable and set to an object literal. So we get excess property checking and checking of the types on the singular function we actually define. I have to as cast it later to the non-partial service, but that seems fine. jest doesn't seem to be able to stub a typescript interface which is why I went that route.